pkg/bucketclass/reconciler.go

```diff
@@ -447,3 +454,21 @@ func (r *Reconciler) UpdateBucketClass(bucketNames []string) error {
 	log.Infof("✅ Successfully updated bucket class %q", r.BucketClass.Name)
 	return nil
 }
+
+func isStoreCompatible(bucketClass *nbv1.BucketClass, obj client.Object) bool {
+	if bucketClass.GetNamespace() == obj.GetNamespace() {
+		return true
+	}
+
+	if obj.GetAnnotations()["allowedNamespaces"] != "" {
+		allowedNamespaces := strings.Split(obj.GetAnnotations()["allowedNamespaces"], ",")
+
+		for _, allowedNamespace := range allowedNamespaces {
```

Comment:
Got it ✔️

Reply:
@romayalon I just realized that if `Contains` from utils will be used then a test will start failing. Because in this loop I am also trimming out whitespace (eg: `"allowedNamespaces": "test1,test2, test3,"`) while `Contains` simply compares the slice values and the parameter passed.

1. Should I modify the `allowedNamespaces` slice and then pass it to the `Contains` method? In this case it would be redundant to invoke the `Contains` then I think.
2. Should I leave it as it is?
3. Should I change `Contains` signature to `func (string, []string, ...func(string, string) bool))` so that a custom comparison function can be passed to it without changing the signature at the other places?
4. Should I change `Contains` signature to `func (string, []string, func(string, string) bool))` so that a custom comparison function can be passed to it but other places will also be effected where the function is being invoked?

Reply:
can you trim it inside Contains?

Reply:
@romayalon yes I think we can. I checked all the invocation of the function, it doesn't seem like making this change would break anything.